lib/src/Base/Optim/MultiStart.cxx

```diff
@@ -0,0 +1,212 @@
+/**
+ *  @brief MultiStart optimization algorithm
+ *
+ *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/MultiStart.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Cobyla.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/MonteCarloExperiment.hxx"
+#include "openturns/ComposedDistribution.hxx"
+#include "openturns/Uniform.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(MultiStart);
+
+static const Factory<MultiStart> Factory_MultiStart;
+
+/* Constructor with parameters */
+MultiStart::MultiStart()
+  : OptimizationSolverImplementation()
+  , solver_(new Cobyla)
+  , startingPointNumber_(ResourceMap::GetAsUnsignedInteger("MultiStart-DefaultStartingPointNumber"))
+{
+}
+
+
+/* Constructor with parameters */
+MultiStart::MultiStart(const OptimizationSolver & solver)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+  , startingPointNumber_(ResourceMap::GetAsUnsignedInteger("MultiStart-DefaultStartingPointNumber"))
+{
+  OptimizationProblem problem(getProblem());
+  if (!problem.hasBounds())
+    throw InvalidArgumentException(HERE) << "Cannot use default multi-start without bounds when the experiment is not defined";
+
+  // sample uniformly over the bounds
+  const UnsignedInteger dimension = problem.getDimension();
+  ComposedDistribution::DistributionCollection coll(dimension);
+  for (UnsignedInteger j = 0; j < dimension; ++ j)
+  {
+    coll[j] = Uniform(problem.getBounds().getLowerBound()[j], problem.getBounds().getUpperBound()[j]);
+  }
+  Distribution initialDistribution = ComposedDistribution(coll);
+  experiment_ = MonteCarloExperiment(initialDistribution, 1);
+}
+
+
+/* Constructor with parameters */
+MultiStart::MultiStart(const OptimizationSolver & solver,
+                       const Experiment & experiment)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+  , experiment_(experiment)
+  , startingPointNumber_(ResourceMap::GetAsUnsignedInteger("MultiStart-DefaultStartingPointNumber"))
+{
+}
+
+
+/* Check whether this problem can be solved by this solver.  Must be overloaded by the actual optimisation algorithm */
+void MultiStart::checkProblem(const OptimizationProblem & problem) const
+{
+  // Nothing to do
+}
```

Comment:
@regislebrun solver_.checkProblm does the work
However I do agree Multistart::checkProblem should be explicitely defined (solver_.checkProblem)